PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/RootLayoutController.java

```diff
@@ -137,4 +138,37 @@ private void initializeSelectionManager() {
       }
     });
   }
+
+  @Override
+  public void filesLoaded(InputStream treeFile, InputStream graphFile, InputStream metadataFile) {
+    try {
+      // Note: not really enjoying pro's of factories with this structure.
+      // consider to decouple a bit more, as the factories can't be mocked right now.
+      GraphFactory graphFactory = new FileGraphFactory(graphFile);
+      TreeFactory treeFactory = new FileTreeFactory(treeFile);
+
+      SequenceGraph graph = graphFactory.getGraph();
+      Tree tree = treeFactory.getTree();
+
+      if (graph != null && tree != null) {
+        loadGraph(graph);
+        loadTree(tree);
+      } else {
+        Logger.getLogger(RootLayoutController.class.getName()).log(
+            Level.SEVERE,
+            "tree or graph was null");
+      }
+    } catch (IOException e) {
+      Logger.getLogger(getClass().getName()).log(Level.SEVERE, null, e);
+    }
+  }
+
+  @FXML
+  @SuppressWarnings("unused")
```

Comment:
Maybe we can call this method from here?

With this the SupressWarnings can go too.